### ISSUE TITLE
doc and docx documents do not have header and footer data

### ISSUE THREAD

**Issue by sxj2024** [microfocus-idol/idol-oem-tutorials#12]
When reading Office documents, I can't extract the headers and footers. Is this configurable, or is there an interface that can achieve this?

**Comment by brian-cowe-mf** [comment]
This is a configuration parameter.  See [the programming guide](https://www.microfocus.com/documentation/idol/IDOL_24_3/KeyviewFilterSDK_24.3_Documentation/Guides/html/cpp-programming/Content/C++/reference/ConfigurationClass/header_and_footer.htm).

**Comment by sxj2024** [comment]
I'm using the C language SDK, but I haven't found any related settings. Is there a documentation link

**Comment by brian-cowe-mf** [comment]
Yes, and the equivalent parameter is available in C.  See [the programming guide for C](https://www.microfocus.com/documentation/idol/IDOL_24_3/KeyviewFilterSDK_24.3_Documentation/Guides/html/c-programming/Content/C/filtering_structures/KVFilterInitOptions.htm#Flags-for-dwFlag).  We support C, C++, Java, .NET and Python, and all of the programming guides can be found [here](https://www.microfocus.com/documentation/idol/keyview_24.3/).

**Comment by sxj2024** [comment]
I carefully checked the C API documentation link you provided, but I can't access it. Could you check if the address has a problem? Also, does the C SDK prior to version 24.4 not support extracting header and footer data from documents? Additionally, if I want to implement a new parser myself, is there a complete tutorial available? Thank you for your help!

**Comment by brian-cowe-mf** [comment]
Ah sorry, I have now corrected the deep link.  The parameter you need is KVF_HEADERFOOTER.  The link for the complete set of guides is correct.

If you're looking to detect or filter a currently undetected or unfilterable format, the first port of call would be to discuss it with the technical support team by raising a support ticket.  There may already be upcoming plans in place to implement it, or this might be the additional customer request that gets what you need prioritised.
